src/EventStore.Client.Streams/EventStoreClient.Append.cs

```diff
@@ -138,40 +138,47 @@ public partial class EventStoreClient {
 						header.Options.StreamIdentifier, writeResult.LogPosition, writeResult.NextExpectedStreamRevision);
 				} else {
 					if (response.WrongExpectedVersion != null) {
-						var currentRevision = response.WrongExpectedVersion.CurrentRevisionOptionCase switch {
-							AppendResp.Types.WrongExpectedVersion.CurrentRevisionOptionOneofCase.NoStream =>
+						var actualStreamRevision = response.WrongExpectedVersion.CurrentRevisionOptionCase switch {
+							AppendResp.Types.WrongExpectedVersion.CurrentRevisionOptionOneofCase.CurrentNoStream =>
 							StreamRevision.None,
 							_ => new StreamRevision(response.WrongExpectedVersion.CurrentRevision)
 						};
 
 						_log.LogDebug(
 							"Append to stream failed with Wrong Expected Version - {streamName}/{expectedRevision}/{currentRevision}",
-							header.Options.StreamIdentifier, new StreamRevision(header.Options.Revision), currentRevision);
-						
+							header.Options.StreamIdentifier, new StreamRevision(header.Options.Revision), actualStreamRevision);
+
 						if (operationOptions.ThrowOnAppendFailure) {
-							if (header.Options.ExpectedStreamRevisionCase ==
-							    AppendReq.Types.Options.ExpectedStreamRevisionOneofCase.Revision) {
+							if (response.WrongExpectedVersion.ExpectedRevisionOptionCase == AppendResp.Types
+								.WrongExpectedVersion.ExpectedRevisionOptionOneofCase.ExpectedRevision) {
 								throw new WrongExpectedVersionException(header.Options.StreamIdentifier,
-									new StreamRevision(header.Options.Revision),
-									currentRevision);
+									new StreamRevision(response.WrongExpectedVersion.ExpectedRevision),
+									actualStreamRevision);
 							}
 
-							var streamState = header.Options.ExpectedStreamRevisionCase switch {
-								AppendReq.Types.Options.ExpectedStreamRevisionOneofCase.Any =>
+							var expectedStreamState = response.WrongExpectedVersion.ExpectedRevisionOptionCase switch {
+								AppendResp.Types.WrongExpectedVersion.ExpectedRevisionOptionOneofCase.ExpectedAny =>
 								StreamState.Any,
-								AppendReq.Types.Options.ExpectedStreamRevisionOneofCase.NoStream =>
+								AppendResp.Types.WrongExpectedVersion.ExpectedRevisionOptionOneofCase.ExpectedNoStream =>
 								StreamState.NoStream,
-								AppendReq.Types.Options.ExpectedStreamRevisionOneofCase.StreamExists =>
+								AppendResp.Types.WrongExpectedVersion.ExpectedRevisionOptionOneofCase.ExpectedStreamExists =>
 								StreamState.StreamExists,
 								_ => throw new InvalidOperationException()
 							};
 
 							throw new WrongExpectedVersionException(header.Options.StreamIdentifier,
-								streamState, currentRevision);
+								expectedStreamState, actualStreamRevision);
+						}
+
+						if (response.WrongExpectedVersion.CurrentRevisionOptionCase == AppendResp.Types
+							.WrongExpectedVersion.CurrentRevisionOptionOneofCase.CurrentRevision) {
+							writeResult = new WrongExpectedVersionResult(header.Options.StreamIdentifier,
```

Comment:
@thefringeninja  Was the new `WrongExpectedVersionResult` constructor that also takes `StreamRevision actualStreamRevision` as parameter meant to be used here and if the `else` branch as well?

Reply:
It looks like it should be:
```
writeResult = new WrongExpectedVersionResult(header.Options.StreamIdentifier,
								new StreamRevision(response.WrongExpectedVersion.ExpectedRevision), actualVersion);
```

Reply:
It also looks like the if condition should be as follows:
```
						if (response.WrongExpectedVersion.ExpectedRevisionOptionCase == AppendResp.Types
							.WrongExpectedVersion.ExpectedRevisionOptionOneofCase.ExpectedRevision) {
```

Reply:
After testing with the above changes, it seems that we also need an `else if` branch to check for: `ExpectedStreamExists` which would set an expected stream revision of -4